src/main/java/com/ebivariation/contigalias/dao/FTPClientAssemblyDaoImplement.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2020 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.ebivariation.contigalias.dao;
+
+import com.ebivariation.contigalias.dus.AssemblyReportReader;
+import com.ebivariation.contigalias.dus.NCBIBrowser;
+import com.ebivariation.contigalias.entities.AssemblyEntity;
+import org.springframework.stereotype.Repository;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Optional;
+
+import static com.ebivariation.contigalias.dus.NCBIBrowser.PATH_GENOMES_ALL;
+
+@Repository("ftpDao")
+public class FTPClientAssemblyDaoImplement implements AssemblyDao {
+
+    @Override
+    public Optional<AssemblyEntity> getAssemblyByAccession(String accession) throws IOException {
+        NCBIBrowser ncbiBrowser = new NCBIBrowser();
+        ncbiBrowser.connect();
+        Optional<String> directory = ncbiBrowser.getGenomeReportDirectory(accession);
+        if (directory.isEmpty()) {
+            return Optional.empty();
+        }
+        AssemblyEntity assemblyEntity;
+        try (InputStream stream = ncbiBrowser.getAssemblyReportInputStream(PATH_GENOMES_ALL + directory)) {
```

Comment:
if we have a NCBIBrowser, the DAO should not need to know about the PATH_GENOMES_ALL, as that is a detail about the FTP. the `PATH_GENOMES_ALL + directory` should be inside the ncbiBrowser.getGenomeReportDirectory method.

Reply:
Fixed